### PR TITLE
[WIP] Ensure Deprecation warnings end up in logs (not stderr)

### PR DESCRIPTION
**Pull request by dims**
Change-Id: If3ccf6cf9ec75233fbbbe3b5d91f89661ad9e531

**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Comment by lavalamp** [comment]
It seems like the actual problem is that the flagset isn't hooked up to the correct error logging mechanism?

The bit about deprecation warnings in the title threw me off. That just happens to be something that the flagset logs, but anything it logged would have this problem. Right?

Assuming I understand it right, this LGTM

**Comment by dims** [comment]
yep, anything logged by flagset would have this problem.

File: pkg/kubectl/cmd/util/helpers.go [hunk 1]
```diff
@@ -660,3 +661,12 @@ func Warning(cmdErr io.Writer, newGeneratorName, oldGeneratorName string) {
...
 }
+
+// WarningWriter serves as a bridge between the standard log package and the klog package.
+type WarningWriter struct{}
+
+// Write implements the io.Writer interface.
+func (writer WarningWriter) Write(data []byte) (n int, err error) {
+	klog.WarningDepth(1, string(data))
+	return len(data), nil
+}
```
Comment (neolit123) [inline pkg/kubectl/cmd/util/helpers.go]: can we use WarningWriter/KlogWriter from component-base for cmd/kubectl too?

**Comment by neolit123** [comment]
cc @mtaufen 
we spoke about this problem briefly during last week's WG-component-standard meeting.

**Review by vincepri — APPROVED** [review]
/lgtm

File: cmd/controller-manager/app/options/kubecloudshared.go [hunk 2]
```diff
@@ -49,6 +50,7 @@ func (o *KubeCloudSharedOptions) AddFlags(fs *pflag.FlagSet) {
 		return
 	}
 
+	fs.SetOutput(logs.WarningWriter())
```
Comment (mtaufen) [inline cmd/controller-manager/app/options/kubecloudshared.go]: Is everything a FlagSet logs suitable for the Warning level?

Comment (dims) [reply cmd/controller-manager/app/options/kubecloudshared.go]: @mtaufen i think so . it was using `os.Stderr` so seemed fair to use `Warning` for those

File: pkg/kubectl/cmd/util/helpers.go (hunk 1 above)
Comment (dims) [reply pkg/kubectl/cmd/util/helpers.go]: @neolit123 that was what i tried first, looks like stuff was copied out to separate out kubectl and importing component-base causes double-ly registered flags now. so i had to have an extra copy

File: cmd/controller-manager/app/options/kubecloudshared.go (hunk 2 above)
Comment (mtaufen) [reply cmd/controller-manager/app/options/kubecloudshared.go]: `os.Stderr` could be used for warnings, errors, or fatal errors. Can you double-check?

**Comment by lavalamp** [comment]
Ping me when you're ready for an approve, I probably won't be able to monitor my inbox too closely the rest of this week.

File: pkg/kubectl/cmd/util/helpers.go (hunk 1 above)
Comment (neolit123) [reply pkg/kubectl/cmd/util/helpers.go]: seems like a problem kubectl/component-base need to resolve.

**Comment by dims** [comment]
found larger issue we need to deal with https://github.com/kubernetes/kubernetes/issues/77305 we need to wait for that to get resolved before we pick this thread up again

**Pull request closed without merging**